phpBB/posting.php

```diff
@@ -616,6 +616,7 @@
 if ($mode != 'edit')
 {
 	$post_data['enable_sig']		= ($config['allow_sig'] && $user->optionget('attachsig')) ? true: false;
+	$post_data['topic_status']		= ($mode == 'post' && (isset($_POST['lock_topic']) && $auth->acl_get('m_lock', $forum_id))) ? true : false;
```

Comment:
Move this to its own `if ($mode == 'post')` block outside of the `if ($mode == 'edit')`. The way it is now is causing the "Lock topic" checkbox to be unset when replying/quoting in a locked topic (as a moderator).